library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/processor/RecordedDataProcessor.kt

```diff
@@ -187,6 +187,16 @@ internal class RecordedDataProcessor(
     ): Runnable? {
         val rumContextData = rumContextDataHandler.createRumContextData() ?: return null
 
+        updateViewSent(rumContextData.newRumContext)
+
+        return runnableFactory(
```

Comment:
the whole idea about runnable here was to be able to move some operations outside of the `WorkerThread` like the timestamp and rum context creations. Right now all these operations are in your `BlockingQueueHandler` and when you reach this method you are already in the `WorkerThread`. You will not need any runnable here and we should remove this part to simplify this class.

Reply:
doesn't timestamp need to be on the uithread to avoid skew?

Reply:
actually this part is only relevant to the touch actions and doesn't relate to the snapshots so it isn't going through the blockinghandler and still follows the original logic (with the exception that runcontextdata has been extracted to its own class)

Reply:
Hmm what timestamp ? The way I see it you are passing the `timestamp` from the `BlockingQueueItem` which was added in the `MainThread`. When you reach the `RecordedDataProcessor#processScreenSnapshots` you are already in the `WorkerThread` as you call this from your executor in the `BlockingQueueHandler` so everything you do in this class is already in the `WorkerThread`

Reply:
I see, well this part is a bit worrying as we are mixing things a bit here. Actually if you delay the snapshots but execute the touch actions you will have issues in the player. Also we are mixing a bit too much things in here. Instead I would assess how we can add the touch actions into the blocking queue also and then have the processor only working with the queue. Will allow you to remove a lot of complexity from here and get rid of executor service and runnables.

Reply:
I've added touch events to the flow